x/shield/handler.go

```diff
@@ -178,63 +161,59 @@ func handleMsgResumePool(ctx sdk.Context, msg types.MsgResumePool, k Keeper) (*s
 	return &sdk.Result{Events: ctx.EventManager().Events()}, nil
 }
 
-func handleMsgPurchaseShield(ctx sdk.Context, msg types.MsgPurchaseShield, k Keeper) (*sdk.Result, error) {
-	_, err := k.PurchaseShieldForP(ctx, msg.PoolID, msg.Shield, msg.Description, msg.From)
-	if err != nil {
+func handleMsgWithdrawForeignRewards() (*sdk.Result, error) {
+	return &sdk.Result{}, types.ErrOperationNotSupported
+}
```

Comment:
why not just remove the entire msg type?

Reply:
Removed.